compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -756,6 +756,44 @@ impl<'a> Parser<'a> {
             }
         }
 
+        // We could've continued to parse bounds but we found a type instead of a trait.
+        if self.may_recover() && !self.token.can_begin_bound() && self.token.can_begin_type() {
+            if let Ok(ty) = self.parse_ty() {
```

Comment:
Aren't you dropping a `Diagnostic` on the floor here? That's gonna cause an ICE. You can change this to either use `?` or `match`.

Reply:
I don't quite understand?

Reply:
Right, since you use `match` which I suggested, you need to cancel the error, i.e. add `Err(err) => err.cancel()`.

Reply:
No, we still want to emit the error even if we can't recover. Better this error than some random "expected `!` but found `,`"; recovery is just to prevent another error later on (not the end of the world if it's emitted).

Reply:
```suggestion
            let ty = self.parse_ty()?;
```